devito/dimension.py

```diff
@@ -348,6 +348,15 @@ class ConditionalDimension(DerivedDimension):
         * ``condition``: an arbitrary SymPy expression depending on ``parent``.
                          All iterations for which the expression evaluates to
                          True are part of the ``SubDimension`` region.
+
+    ConditioanlDimension needs runtime arguments. The generated C code will require
+    the size of the dimension to initialize the arrays as e.g:
+    .. code-block:: python
```

Comment:
if you don't leave blank lines and indent it properly, I'm not sure this is gonna be rendered properly (I might be wrong). For example:

```
ConditionalDimension needs runtime arguments. The generated C code will require
the size of the dimension to initialize the arrays as, for example:

    .. code-block:: python

        <code goes here>
```

Can you check that?